server/src/main/java/org/apache/iotdb/db/conf/directories/DirectoryManager.java

```diff
@@ -64,11 +65,53 @@ private DirectoryManager() {
       strategyName = IoTDBDescriptor.getInstance().getConfig().getMultiDirStrategyClassName();
       Class<?> clazz = Class.forName(strategyName);
       sequenceStrategy = (DirectoryStrategy) clazz.newInstance();
-      sequenceStrategy.init(sequenceFileFolders);
+      sequenceStrategy.setFolders(sequenceFileFolders);
       unsequenceStrategy = (DirectoryStrategy) clazz.newInstance();
-      unsequenceStrategy.init(unsequenceFileFolders);
+      unsequenceStrategy.setFolders(unsequenceFileFolders);
+    } catch (DiskSpaceInsufficientException e) {
+      logger.error("All disks of folders are full.", e);
     } catch (Exception e) {
-      logger.error("can't find sequenceStrategy {} for mult-directories.", strategyName, e);
+      logger.error("Can't find strategy {} for mult-directories.", strategyName, e);
+    }
+  }
+
+  public void updateFileFolders() throws LoadConfigurationException {
+    try {
+      List<String> sequenceFileFolders =
+          new ArrayList<>(Arrays.asList(IoTDBDescriptor.getInstance().getConfig().getDataDirs()));
```

Comment:
Why not `List<String> sequenceFileFolders =
          Arrays.asList(IoTDBDescriptor.getInstance().getConfig().getDataDirs());`?

Reply:
I just imitate line 47 in this class. I have fixed them all.